src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -784,6 +866,28 @@ void ov::npuw::LLMCompiledModel::convert_stateful_lora_to_stateless(std::shared_
     model->add_parameters(new_parameters);
 }
 
+void ov::npuw::LLMCompiledModel::gemma_transformations(const std::shared_ptr<ov::Model>& model) {
+    ov::pass::GraphRewrite rewr;
+    auto RewrRes = std::make_unique<GemmaSlidingMask::Result>();
+    rewr.add_matcher<GemmaSlidingMask>(RewrRes.get());
+    rewr.run_on_model(model);
+
+    if (RewrRes->found) {
+        OPENVINO_ASSERT(
+            RewrRes->window_size > 0,
+            "Gemma sliding window size must be strictly positive, but got " + std::to_string(RewrRes->window_size));
+
+        m_gemma_sliding_window_size = RewrRes->window_size;
+        auto mask_input = RewrRes->mask_input;
+        model->add_parameters({mask_input});
+        for (auto&& input : model->inputs()) {
+            if (input.get_node() == mask_input.get()) {
+                input.set_names({mask_input->get_friendly_name()});
+            }
+        }
+    }
```

Comment:
Suggest to do `model->validate_nodes_and_infer_types();` after

Reply:
Added